src/core/server/elasticsearch/elasticsearch_service.ts

```diff
@@ -18,16 +18,26 @@
  */
 
 import { ConnectableObservable, Observable, Subscription } from 'rxjs';
-import { filter, first, map, publishReplay, switchMap } from 'rxjs/operators';
+import { filter, first, map, publishReplay, switchMap, take } from 'rxjs/operators';
 import { merge } from 'lodash';
 import { CoreService } from '../../types';
 import { CoreContext } from '../core_context';
 import { Logger } from '../logging';
 import { ClusterClient } from './cluster_client';
+import { ScopedClusterClient } from './scoped_cluster_client';
 import { ElasticsearchClientConfig } from './elasticsearch_client_config';
 import { ElasticsearchConfig, ElasticsearchConfigType } from './elasticsearch_config';
 import { HttpServiceSetup, GetAuthHeaders } from '../http/';
 
+declare module '../../server' {
```

Comment:
The [Handler RFC](https://github.com/elastic/kibana/blob/master/rfcs/text/0003_handler_interface.md) defines a convention where all core services are under the `core` context name. This is different than what you're doing here.

One one hand, it provides some consistency and implies that all things under `core` are going to be present for all plugins. OTOH, it may be an extra level of nesting that isn't particularly useful. Using a single context provider for all of core also forces us to register this in the Server/CoreSystem classes.

Are you proposing we change this convention? If we decide to do this, we should also update the RFC document and make sure the ApplicationService matches.

Reply:
forgot, updated